### PR TITLE
Hot reload the build captain csv 🔥

### PR DESCRIPTION
**Pull request by bobcatfish**
# Changes

While updating build cop -> build captain, I noticed (which I think
other folks have been saying all along!) that the list of ppl in the csv
is loaded once on startup, meaning if there are any updates, you have to
restart to bot. With this change, the list will be loaded every time it
is accessed.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/meow
/cc @afrittoli @sbwsg

File: bots/buildcaptain/cmd/buildcaptain/main.go [hunk 1]
```diff
@@ -85,10 +81,10 @@ func main() {
 	}
 }
 
-func handleMessage(rtm *slack.RTM, message, channel string, direct bool) {
+func handleMessage(rtm *slack.RTM, message, channel string, direct bool, captainsID map[string]string) {
 	switch {
 	case statusMessage(message, botID, direct):
-		rtm.SendMessage(rtm.NewOutgoingMessage(fmt.Sprintf("<@%s> is the Build Captain :female-pilot:\nBuild Captain log is here: https://docs.google.com/document/d/1kUzH8SV4coOabXLntPA1QI01lbad3Y1wP5BVyh4qzmk", currentCaptain), channel))
+		rtm.SendMessage(rtm.NewOutgoingMessage(fmt.Sprintf("<@%s> is the Build Captain :female-pilot:\nBuild Captain log is here: https://docs.google.com/document/d/1kUzH8SV4coOabXLntPA1QI01lbad3Y1wP5BVyh4qzmk", getCurrentCaptain(captainsID)), channel))
```
Comment (sbwsg) [inline bots/buildcaptain/cmd/buildcaptain/main.go:87]: Am I understanding right that every time I request "status" from the slack bot we're going to make a call to github?

That seems like a vector for a bad actor to take advantage of?  It would suck for our dogfooding cluster's IP addresses to be banned by github because someone thinks it's funny to request the status a million times.

Could we do a best-of-both worlds type thing and request the raw data on a daily or hourly basis instead of every query?

Comment (bobcatfish) [reply bots/buildcaptain/cmd/buildcaptain/main.go:87]: yep that's right!

@sbwsg in this scenario is someone coming into the build cop channel and posting repeatedly? or are they doing this some other way? (i dont think this service would respond to requests that dont come from slack?) would github ban us or would be be temporarily throttled?

if this IS the scenario, im wondering if we'd have a bigger problem - specifically a malicious person in slack

Comment (sbwsg) [reply bots/buildcaptain/cmd/buildcaptain/main.go:87]: I don't totally agree that a user acting badly in slack is a bigger problem than having our dogfooding cluster rate limited by github.  The fallout from a bad user in slack is an unpleasant slack experience.  The fallout for dogfooding is that all of our PRs no longer merge because the checks we have in dogfooding are blocked.

More generally I'm a little puzzled why we wouldn't write this defensively.

Comment (bobcatfish) [reply bots/buildcaptain/cmd/buildcaptain/main.go:87]: > The fallout for dogfooding is that all of our PRs no longer merge because the checks we have in dogfooding are blocked.

I'm trying to understand what would happen if we did get rate limited - as far as i can tell the requests this bot makes are unauthenticated, so the limit is pretty low (https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting) - I'm not sure if we make unauthenticated requests in the rest of our dogfooding (probably we shouldn't given the low rate limiting?), and even if we do, I'm not sure if github would treat these as all coming from the same unauthenticated source:

> Unauthenticated requests are associated with the originating IP address, and not the user making requests.

It seems like nodes in dogfooding have public IPs, so I suppose a node could get rate limited.

But this is only a problem if our CI jobs are also unauthenticated:

> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. 

This would mean our CI would also be limited to 60 requests per hour (per node) :O (and actually means we could get locked out by someone opening too many PRs too quickly 😅 )

So long story short:
1. We should probably make sure we are making authenticated requests in our CI
2. If you really want I can make it so only the daily update reads the file but I honestly don't think this is that big of a risk (certainly less of a risk than what our CI is already doing)
